File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;
        for (int i = 0; i < min((int)nums.size(), k); i++) {
            pq.push(nums[i]);
        }
        for (int i = k; i < (int)nums.size(); i++) {
            if (pq.top() < nums[i]) {
                pq.pop();
                pq.push(nums[i]);
            }
        }
        return pq.top();
    }
};

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        n = len(nums) 
        h = []
       
        for i in range(k):
            heapq.heappush(h, nums[i])

        for i in range(k, n):
            if nums[i] > h[0]:
                heapq.heappushpop(h, nums[i])
            
        return h[0]
        
